### DISCUSSION TITLE
Podman random 502 (issue #2608) solved in your fork ?

### DISCUSSION THREAD

**Discussion by L33thium** [ZoeyVid/NPMplus#289] [Q&A] [answered]
Hi,
The original NPM (jc21) run into troubles especially when using containers hostnames to redirect hosts.
We get random 502 error (approx 1% requests if we redirect to host LAN and 20 to 30% to podman network by hostnames)
Without any traces in the log.

Have you known this issue and find a way to solve it ?

Thanks

**Comment by Zoey2936** [comment, marked as answer]
No this issue is unknown to be, but it can't affect me, since I use normal docker, but yes I would like to support podman, but I think this is an internal problem of networking in podman/nginx. If podman does support network mode host, what about running the npm in this mode (like I recommend) and then exposing your containers to localhost (`127.0.0.1:67:8080` to expose port 8080 of container to localhost 127.0.0.1 in compose or `-p 127.0.0.1:67:8080` in docker run) and after that you could add your container to the npm by using 127.0.0.1 as ip and 67 as port and http/https as your container needs it in reverse proxy mode?

**Reply by L33thium** [reply]
Thank you for your quick answer, 
i know the workaround and it's more or less what i do, but for security considerations i prefer to keep containers confined with only one controled exit throught the proxy.
So you confirmed it's probably an issue beetween podman and nginx, i have to found an other solution.

Thanks again

**Reply by Zoey2936** [reply]
Can you maybe share this solution for others with a similar problem?

**Reply by L33thium** [reply]
Not a solution, a workaround, the container exposes a port to the host, the proxy forward to the lan address of the host, to the container port.

**Reply by Zoey2936** [reply]
I would mostly say, the problem is caused by podman with nothing I would do...

**Reply by L33thium** [reply]
it's beetween podman and nginx.
Podman is a lot differentl than Docker (but it's very compatible so it may be disturbing).
The docker daemon provides networking and name resolution by itself.
There is no Podman daemon netxorking and name resolution is provided by the Linux kernel, netavark and aardvark.

I will try traefik, acting differently, i guess it's using the orchestrator API to resolve containers names, not DNS requests
